src/main/scala/lms/transformation/mpi_nccl_transformation/distributedTensor2MPINCCLConv.scala

```diff
@@ -236,16 +236,121 @@ trait DistributeTensor2MPI_NCCLConv extends DistributeTensor2MPI_NCCLBase with C
         convAlgo, workspace_bytes_v))
       generate_comment("end finding convolution backward filter workspace size")
 
-      generate_comment("begin allocating gpu array for onvolution backward filter workspace")
-      val d_workspace = gpu_array1(INT(workspace_bytes_v(pos)), manifest[Float], myNCCLRank)
-      generate_comment("end allocating gpu array for onvolution backward filter workspace")
+      generate_comment("begin allocating gpu array for convolution backward filter workspace")
+      val d_workspace = gpu_array1_bytes(INT(workspace_bytes_v(pos)), manifest[Float], myNCCLRank)
+      generate_comment("end allocating gpu array for convolution backward filter workspace")
      
       generate_comment("begin convolution backward filter pass")
       CUDNN_CHECK(CUDNN_CONV_BWD_FILTER(myCUDNNComm, VAR(FLOAT(alpha)), weight_descriptor, new ARRAY(weight_tensor), doutput_descriptor, new ARRAY(doutput_tensor),
         conv_descriptor, convAlgo, d_workspace, workspace_bytes_v, VAR(FLOAT(beta)), dfilter_descriptor, dfilter))
       generate_comment("end convolution backward filter pass")
 
       dfilter.x
+    
+    case Node(s, "tensors_dropout", Backend.Const(tts: List[TensorType])::Backend.Const(anno:Anno)::(input:Backend.Sym)::
+      Backend.Const(params)::_, _) =>
+      implicit val pos = Adapter.oldSourceMap(s)
+
+      // unpack dropout paratemers
+      val DropoutParam(dropout, seed) = params.asInstanceOf[DropoutParam]
+
+      // get input info and transform input tensors
+      val input_shape = tensor_shape(input, useOldMetadata = true)
+      val input_tensor = get_operand(input, anno)
+
+      val input_descriptor = getTensorDescriptor(input_shape, "tensor")
+      val output_descriptor = getTensorDescriptor(input_shape, "tensor")
+
+      generate_comment("begin finding dropout forward reserve space bytes") 
+      var reserve_bytes = 0
+      var reserve_bytes_v = VAR(SIZE_T(reserve_bytes))  // var read
```

Comment:
what if we just do `val reserve_bytes_v = VAR(SIZE_T(0))`?

Reply:
yeah reserve_bytes are not used.

Reply:
fixed in 9ea513c